### PR TITLE
Supporting infrastructure and further unit tests for the default R packages

### PR DESCRIPTION
**Pull request by georgestagg**
This PR adds a modified version of `tools::testInstalledPackage` to the webr R package. The function takes a package's examples and test files and runs them, checking to see if any uncaught errors were raised.

The R test code is run in the current process using `source`, rather than launching an independent R session with `system`. Since we are not in a subprocess, some calls to `quit` are removed in the test code before running. Minor changes in output between using `source` and batch running R code means that we cannot easily compare exact test output to expected values, so this is skipped. Similar to the version in `tools`, success is denoted by returning 0.

In the future, when we can launch an R subprocess, we can probably remove this `test_package` function and use the provided function in `tools` directly instead.

Some examples trigger prompts when R is in interactive mode, for example asking the user to press Enter to continue plotting. I could not see an R launch argument that forces starting in _non_-interactive mode, so I have added a webR launch option that sets the `R_Interactive` global after `init` has finished. 

The testing is done by running `webr::test_package` through `webR.evalRCode`, after launching R with `interactive=false`, using the jest testing framework that's now in place.

As far as I can tell, a lot of examples and tests from the default R packages seem to run OK. I have been mostly testing in Node, but they can also be run in the browser through the REPL. Some package tests fail in the browser but work OK in Node - for example, the `utils` package fails in the browser attempting to make a cross-origin network request.

There are still a few issues:
 * Several examples and tests from various default packages do fail. I have disabled or worked around the failing tests with R source patches for now.
 * Since everything is run in the same R process, tests can interfere with each other.
 * The `tcltk` package is not supported with a lack of Tcl/Tk. I have removed the package from VFS for now - although we could perhaps find a Tk library for WASM in future.
 * The `parallel` package is not supported as we can't `fork` or create subprocesses yet. I have removed the package from the VFS for now.

### CODE REVIEW

File: packages/webr/R/testing.R [hunk 1]
```diff
@@ -0,0 +1,90 @@
+# Run a R script within in this instance of R, capturing output to a file.
+sink_source_to_file <- function(sourcefile, outfile) {
```
Comment (lionel-) [inline packages/webr/R/testing.R:2]: Style nit:

```suggestion
sink_source_to_file <- function(src_file, out_file) {
```

File: packages/webr/R/testing.R [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+        return(invisible(1L))
+      } else {
+        file.rename(failfile, outfile)
+      }
+    } else {
+      warning(gettextf("no examples found for package %s", sQuote(pkg)),
+        call. = FALSE, domain = NA)
+    }
+
+    if (dir.exists(d <- file.path(pkgdir, "tests"))) {
```
[the remark this answers is not in the document]
Comment (lionel-) [reply packages/webr/R/testing.R:67]: Though it's fine to stay close to the original source if this implementation is closely related to a base function.

File: packages/webr/R/testing.R [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+        message(gettextf("  Running %s", sQuote(f)), domain = NA)
+        outfile <- sub("rout$", "Rout", paste0(f, "out"))
+        res <- sink_source_to_file(f, outfile)
+        if (res) {
+          file.rename(outfile, paste0(outfile, ".fail"))
+          return(invisible(1L))
+        }
+      }
+    }
+    setwd(owd)
```
Comment (lionel-) [inline packages/webr/R/testing.R:88]: Should be restored on-exit.

**Review by lionel- — COMMENTED** [review]
Some style comments

File: packages/webr/R/testing.R [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+}
+
+# Remove lines from a file on disk. We use this to remove calls to quit(),
+# since we are running examples within the current R session.
+remove_lines_in_file <- function(sourcefile, lines) {
+    con <- file(sourcefile)
+    source <- readLines(con)
+    for (line in lines) {
+      source <- source[source != line]
+    }
```
[the remark this answers is not in the document]
Comment (georgestagg) [reply packages/webr/R/testing.R]: Nice!

File: packages/webr/R/testing.R [hunk 5]
```diff
@@ -0,0 +1,90 @@
...
+    writeLines(source, con)
+    close(con)
```
[the remark this answers is not in the document]
Comment (georgestagg) [reply packages/webr/R/testing.R]: Thanks, I've removed `file` and `close` here. I had misunderstood `readLines` and `writeLines` slightly.

File: packages/webr/R/testing.R (hunk 2 above)
[the remark this answers is not in the document]
Comment (georgestagg) [reply packages/webr/R/testing.R]: I've moved the assignment to just outside the `if`.

File: packages/webr/R/testing.R [hunk 6]
```diff
@@ -0,0 +1,90 @@
+# Run a R script within in this instance of R, capturing output to a file.
+sink_source_to_file <- function(sourcefile, outfile) {
+  r <- 0L
+  con <- file(outfile, open = "wt")
+  sink(con)
+  sink(con, type = "message")
+  tryCatch({
```
[the remark this answers is not in the document]
Comment (georgestagg) [reply packages/webr/R/testing.R]: I've moved the `tryCatch` outside this function and the `finally` part into an `on.exit`.

**Pull request merged**